invisible_cities/io/mcinfo_io.py

```
@@ -0,0 +1,305 @@
+    """Returns all hits in the event"""
+    mchits_dict = {}
+    for event_no, particle_dict in mcevents.items():
+        hits = []
+        for particle_no in particle_dict.keys():
+            particle = particle_dict[particle_no]
+            hits.extend(particle.hits)
+        mchits_dict[event_no] = hits
+    return mchits_dict
+
+
+def read_mcsns_response(h5f, event_range=(0,1e9)) ->Mapping[int, Mapping[int, Waveform]]:
+
+    h5config = h5f.root.MC.configuration
+
+    bin_width_PMT  = 100. * units.nanosecond
+    bin_width_SiPM = 1. * units.microsecond
+    for row in h5config:
+        param_name = row['param_key'].decode('utf-8','ignore')
+        if param_name.find('time_binning') >= 0:
+            param_value = row['param_value'].decode('utf-8','ignore')
+            numb, unit = param_value.split()
+            if param_name.find('Pmt') > 0:
+                bin_width_PMT = float(numb)
+            elif param_name.find('SiPM') >= 0:
+                bin_width_SiPM = float(numb)
+
+    h5extents = h5f.root.MC.extents
+    h5waveforms = h5f.root.MC.waveforms
+
+    last_line_of_event = 'last_sns_data'
+
+    all_events = {}
+
+    iwvf = 0
+    if(event_range[0] > 0):
+        iwvf = h5extents[event_range[0]-1][last_line_of_event]+1
+
+    for iext in range(*event_range):
+        if (iext >= len(h5extents)):
+            break
+
+        current_event = {}
+
+        iwvf_end = h5extents[iext][last_line_of_event]
+        current_sensor_id = h5waveforms[iwvf]['sensor_id']
+        time_bins = []
+        charges = []
+        while (iwvf <= iwvf_end):
+            wvf_row = h5waveforms[iwvf]
+            sensor_id = wvf_row['sensor_id']
+
+            if (sensor_id == current_sensor_id):
+                time_bins.append(wvf_row['time_bin'])
+                charges.append(wvf_row['charge'])
+            else:
+               # current_event[current_sensor_id] = zip(time_bins, charges)
+                if 0 <= sensor_id < 1000:
```

do you have a special case `sensor_id < 0` and do you want to assume it is a PMT? If so, it is ok, but I would raise an exception (`ValueError`, I guess) otherwise.

---

I don't think we actually want to check every possible scenario (`sensor_id`'s between X064 and (X+1)000 would also be fake, for instance). So you can simply rephrase it as `bin_width = bin_width_PMT if sensor_id < 1000 else bin_width_SiPM`.